test/Microsoft.AspNetCore.Mvc.Core.Test/ModelBinding/Binders/EnumTypeModelBinderTest.cs

```diff
@@ -0,0 +1,316 @@
+        [InlineData("Value2, Value8", false)]
+        [InlineData("value8,value4,value2,value1", false)]
+        public async Task BindModel_BindsTo_NonNullableEnumType(string flagsEnumValue, bool allowBindingUndefinedValueToEnumType)
+        {
+            // Arrange
+            var modelType = typeof(FlagsEnum);
+            var enumConverter = TypeDescriptor.GetConverter(modelType);
+            var expected = enumConverter.ConvertFrom(flagsEnumValue).ToString();
+            var binderInfo = GetBinderAndContext(
+                modelType,
+                allowBindingUndefinedValueToEnumType,
+                valueProviderValue: new object[] { flagsEnumValue });
+            var bindingContext = binderInfo.Item1;
+            var binder = binderInfo.Item2;
+
+            // Act
+            await binder.BindModelAsync(bindingContext);
+
+            // Assert
+            Assert.True(bindingContext.Result.IsModelSet);
+            var boundModel = Assert.IsType<FlagsEnum>(bindingContext.Result.Model);
+            Assert.Equal(expected, boundModel.ToString());
+        }
+
+        [Theory]
+        [InlineData("1", true)]
+        [InlineData("8, 1", true)]
+        [InlineData("Value2, Value8", true)]
+        [InlineData("value8,value4,value2,value1", true)]
+        [InlineData("1", false)]
+        [InlineData("8, 1", false)]
+        [InlineData("Value2, Value8", false)]
+        [InlineData("value8,value4,value2,value1", false)]
+        public async Task BindModel_BindsTo_NullableEnumType(string flagsEnumValue, bool allowBindingUndefinedValueToEnumType)
+        {
+            // Arrange
+            var modelType = typeof(FlagsEnum?);
+            var enumConverter = TypeDescriptor.GetConverter(GetUnderlyingType(modelType));
+            var expected = enumConverter.ConvertFrom(flagsEnumValue).ToString();
+            var binderInfo = GetBinderAndContext(
+                modelType,
+                allowBindingUndefinedValueToEnumType,
+                valueProviderValue: new object[] { flagsEnumValue });
+            var bindingContext = binderInfo.Item1;
+            var binder = binderInfo.Item2;
+
+            // Act
+            await binder.BindModelAsync(bindingContext);
+
+            // Assert
+            Assert.True(bindingContext.Result.IsModelSet);
+            var boundModel = Assert.IsType<FlagsEnum>(bindingContext.Result.Model);
+            Assert.Equal(expected, boundModel.ToString());
+        }
+
+        [Theory]
+        [InlineData(typeof(FlagsEnum), "Value10")]
+        [InlineData(typeof(FlagsEnum), "Value1, Value10")]
+        [InlineData(typeof(FlagsEnum), "value10, value1")]
+        [InlineData(typeof(FlagsEnum?), "Value10")]
+        [InlineData(typeof(FlagsEnum?), "Value1, Value10")]
+        [InlineData(typeof(FlagsEnum?), "value10, value1")]
+        public async Task BindModel_AddsErrorToModelState_ForEnumValues_NotValid(Type modelType, string suppliedValue)
```

Comment:
Not mandatory but I suggest &hellip;For**Invalid**EnumValues in this and later tests with similar names.